app/src/main/java/com/github/se/studybuddies/viewModels/DatabaseConnection.kt

```diff
@@ -2,104 +2,179 @@ package com.github.se.studybuddies.viewModels
 
 import android.net.Uri
 import android.util.Log
-import androidx.compose.ui.platform.LocalContext
-import com.github.se.studybuddies.data.User
-import com.google.android.gms.tasks.OnCompleteListener
+import com.github.se.studybuddies.data.Group
+import com.github.se.studybuddies.data.GroupList
 import com.google.android.gms.tasks.Task
 import com.google.firebase.auth.FirebaseAuth
-import com.google.firebase.database.DataSnapshot
-import com.google.firebase.database.DatabaseError
-import com.google.firebase.database.DatabaseReference
-import com.google.firebase.database.FirebaseDatabase
-import com.google.firebase.database.ValueEventListener
 import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FieldValue
 import com.google.firebase.firestore.FirebaseFirestore
-import com.google.firebase.firestore.SetOptions
 import com.google.firebase.storage.FirebaseStorage
-import java.time.ZoneId
-import java.util.Date
 import kotlinx.coroutines.tasks.await
 
 class DatabaseConnection {
-    private val db: FirebaseFirestore = FirebaseFirestore.getInstance()
-    private val storage = FirebaseStorage.getInstance().reference
+  private val db: FirebaseFirestore = FirebaseFirestore.getInstance()
+  private val storage = FirebaseStorage.getInstance().reference
 
-    // all collections
-    private val userDataCollection = db.collection("userData")
+  // all collections
+  private val userDataCollection = db.collection("userData")
+  private val userMembershipsCollection = db.collection("userMemberships")
+  private val groupDataCollection = db.collection("groupData")
 
-    fun getCurrentUserUID(): String {
-        val uid = FirebaseAuth.getInstance().currentUser?.uid
-        return if (uid != null) {
-            Log.d("MyPrint", "Fetched user UID is $uid")
-            uid
-        } else {
-            Log.d("MyPrint", "Failed to get current user UID")
-            ""
-        }
+  // using the userData collection
+  fun getCurrentUserUID(): String {
+    val uid = FirebaseAuth.getInstance().currentUser?.uid
+    return if (uid != null) {
+      Log.d("MyPrint", "Fetched user UID is $uid")
+      uid
+    } else {
+      Log.d("MyPrint", "Failed to get current user UID")
+      ""
     }
+  }
 
-    fun getUserData(uid: String): Task<DocumentSnapshot> {
-        return userDataCollection.document(uid).get()
-    }
-    suspend fun getDefaultProfilePicture(): Uri {
-        return storage.child("userData/default.jpg").downloadUrl.await()
-    }
+  fun getUserData(uid: String): Task<DocumentSnapshot> {
+    return userDataCollection.document(uid).get()
```

Comment:
I coded this way because this is what a TA showed me during the bootcamp. But other teammates would also prefer the DatabaseConnection to return the data struct. However this also changes a lot how I coded the viewModels, so I will be fixing that nice and clean next sprint.